src/controller/markdown.controller.ts

```diff
@@ -0,0 +1,133 @@
+import { marked } from 'marked';
+import { Console } from 'console';
+
+export enum MARKDOWN_DOMAIN {
+  NEWS,
+  UPDATES
+}
+
+export class MarkdownController {
+
+  private static getPathForDomain(domain: MARKDOWN_DOMAIN) {
+    switch (domain) {
+      case MARKDOWN_DOMAIN.NEWS: return Environment.NEWS_PATH;
+      case MARKDOWN_DOMAIN.UPDATES: return Environment.UPDATES_PATH;
+    }
+  }
+  static getAvailableFiles(domain: MARKDOWN_DOMAIN): string[] {
+    return fs.readdirSync(MarkdownController.getPathForDomain(domain)).filter((s) => s.endsWith('.md'));
+  }
+  static getFile(path: string): fs.ReadStream | null {
+    if (fs.existsSync(path)) {
+      return fs.createReadStream(path);
+    }
+    return null;
+  }
+  private static firstLineFs(fileName: string): Promise<string> {
+    const stream = fs.createReadStream(fileName);
+    let acc = '';
+    let pos = 0;
+    let index;
+    return new Promise<string>((resolve, reject) => {
+      stream
+        .on('data', (chunk) => {
+          index = chunk.indexOf('\n');
+          acc += chunk;
+          index !== -1 ? stream.close() : (pos += chunk.length);
+        })
+        .on('close', () => {
+          resolve(acc.slice(0, pos + index));
+        })
+        .on('error', (err) => {
+          reject(err);
+        });
+    });
+  }
+  private static async buildNewsHeadMapping(domain: MARKDOWN_DOMAIN): Promise<[string, string][]> {
+    const newsFiles = MarkdownController.getAvailableFiles(domain);
+    const mapping_p: Promise<[string, string]>[] = newsFiles.map(
+      async (fileName: string) => {
+        const pair: Promise<[string, string]> = MarkdownController.firstLineFs(
+          MarkdownController.getPathForDomain(domain) + '/' + fileName
+        ).then((head) => [fileName, head]);
+        return pair;
+      }
+    );
+    return Promise.all(mapping_p);
+  }
+
+
+  static async availableFilesRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    const payload = MarkdownController.getAvailableFiles(domain);
+    new APIResponse(res, 200, payload).send();
+  }
+  static async fileRequest(req: Request, res: Response, domain: MARKDOWN_DOMAIN): Promise<void> {
+    if (req.params.md === undefined || req.params.md === null)
+      new APIResponse(res, 400, {}, ['Undefined news parameter']).send();
+
+    const path = `${MarkdownController.getPathForDomain(domain)}/${req.params.md}`;
+
+    try {
+      if (fs.existsSync(path)) {
+        const fileContents: Buffer = fs.readFileSync(path);
+        console.log("file loaded", fileContents.toString());
+        const parsedHTML: string = marked.parse(fileContents.toString());
+        console.log("file parsed", parsedHTML);
+        res.set('Content-Type', MIMETYPES.md);
+        res.send(parsedHTML);
+        console.log("file sent");
+      } else {
+        new APIResponse(res, 404, {}, ['File not found']).send();
+      }
+    } catch (e) {
+      console.error("fs error", e);
```

Comment:
Definiere hier besser wo genau auch der error geworfen wird. Also sowas wie das hier:


```suggestion 
      console.error("[MarkDown FileReader] Could not read file: ", e);
```